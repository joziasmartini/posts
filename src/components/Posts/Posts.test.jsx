import { render, screen } from "@testing-library/react";
import { Posts } from ".";

const postsMock = [
  { id: 1, title: "title 01", body: "body 01", cover: "cover 01" },
  { id: 2, title: "title 02", body: "body 02", cover: "cover 02" },
  { id: 3, title: "title 03", body: "body 03", cover: "cover 03" },
];

describe("<Posts />", () => {
  test("should render Posts", () => {
    render(<Posts posts={postsMock} />);

    const titles = screen.getAllByRole("heading", { name: /title/i });
    const covers = screen.getAllByRole("img", { src: /cover/i });
    const bodys = screen.getAllByText(/body/i);

    expect(titles).toHaveLength(3);
    expect(covers).toHaveLength(3);
    expect(bodys).toHaveLength(3);
  });

  test("should match snapshot", () => {
    const { container } = render(<Posts posts={postsMock} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
