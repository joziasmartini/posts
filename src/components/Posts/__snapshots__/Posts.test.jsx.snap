// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="title 01"
      src="cover 01"
    />
    <div
      class="post-content"
    >
      <h2>
        title 01
         
        1
      </h2>
      <p>
        body 01
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="title 02"
      src="cover 02"
    />
    <div
      class="post-content"
    >
      <h2>
        title 02
         
        2
      </h2>
      <p>
        body 02
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="title 03"
      src="cover 03"
    />
    <div
      class="post-content"
    >
      <h2>
        title 03
         
        3
      </h2>
      <p>
        body 03
      </p>
    </div>
  </div>
</div>
`;
